sentry/src/main/java/io/sentry/CombinedScopeView.java

```diff
@@ -431,8 +433,7 @@ public void setPropagationContext(@NotNull PropagationContext propagationContext
 
   @Override
   public @NotNull IScope clone() {
-    // TODO [HSM] just return a new CombinedScopeView with forked scope?
-    return getDefaultWriteScope().clone();
+    return new CombinedScopeView(globalScope, isolationScope.clone(), scope.clone());
```

Comment:
The idea here is to have a clone that doesn't change the original so we're safer by cloning both isolation and current scope.